libs/labelbox/src/labelbox/schema/dataset.py

```diff
@@ -139,82 +138,38 @@ def create_data_row(self, items=None, **kwargs) -> "DataRow":
                 in `kwargs`.
             InvalidAttributeError: in case the DB object type does not contain
                 any of the field names given in `kwargs`.
-
+            ResourceCreationError: If data row creation failed on the server side.
         """
-        invalid_argument_error = "Argument to create_data_row() must be either a dictionary, or kwargs containing `row_data` at minimum"
-
-        def convert_field_keys(items):
-            if not isinstance(items, dict):
-                raise InvalidQueryError(invalid_argument_error)
-            return {
-                key.name if isinstance(key, Field) else key: value
-                for key, value in items.items()
-            }
-
-        if items is not None and len(kwargs) > 0:
-            raise InvalidQueryError(invalid_argument_error)
+        file_upload_thread_count = 1
+        args = items if items is not None else kwargs
+
+        upload_items = self._separate_and_process_items([args])
+        specs = DataRowCreateItem.build(self.uid, upload_items)
+        task: DataUpsertTask = self._exec_upsert_data_rows(
+            specs, file_upload_thread_count)
```

Comment:
It's common for the sync version of a method to simply call the async version and block on the result (to avoid code duplication).

In this case, `create_data_row` can invoke `create_data_rows_sync` and `create_data_rows_sync` can invoke `create_data_rows`.

Also, I would consider deprecating some of these redundant methods.

Reply:
will implement... I can add deprecation warning, need to confirm with the sdk team first